test/intl402/String/prototype/toLocaleUpperCase/special_casing_Lithuanian.js

```diff
@@ -0,0 +1,188 @@
+  "\u2C7C",             // LATIN SUBSCRIPT SMALL LETTER J
+  "\uD835\uDC22", "\uD835\uDC23",   // MATHEMATICAL BOLD SMALL I..MATHEMATICAL BOLD SMALL J
+  "\uD835\uDC56", "\uD835\uDC57",   // MATHEMATICAL ITALIC SMALL I..MATHEMATICAL ITALIC SMALL J
+  "\uD835\uDC8A", "\uD835\uDC8B",   // MATHEMATICAL BOLD ITALIC SMALL I..MATHEMATICAL BOLD ITALIC SMALL J
+  "\uD835\uDCBE", "\uD835\uDCBF",   // MATHEMATICAL SCRIPT SMALL I..MATHEMATICAL SCRIPT SMALL J
+  "\uD835\uDCF2", "\uD835\uDCF3",   // MATHEMATICAL BOLD SCRIPT SMALL I..MATHEMATICAL BOLD SCRIPT SMALL J
+  "\uD835\uDD26", "\uD835\uDD27",   // MATHEMATICAL FRAKTUR SMALL I..MATHEMATICAL FRAKTUR SMALL J
+  "\uD835\uDD5A", "\uD835\uDD5B",   // MATHEMATICAL DOUBLE-STRUCK SMALL I..MATHEMATICAL DOUBLE-STRUCK SMALL J
+  "\uD835\uDD8E", "\uD835\uDD8F",   // MATHEMATICAL BOLD FRAKTUR SMALL I..MATHEMATICAL BOLD FRAKTUR SMALL J
+  "\uD835\uDDC2", "\uD835\uDDC3",   // MATHEMATICAL SANS-SERIF SMALL I..MATHEMATICAL SANS-SERIF SMALL J
+  "\uD835\uDDF6", "\uD835\uDDF7",   // MATHEMATICAL SANS-SERIF BOLD SMALL I..MATHEMATICAL SANS-SERIF BOLD SMALL J
+  "\uD835\uDE2A", "\uD835\uDE2B",   // MATHEMATICAL SANS-SERIF ITALIC SMALL I..MATHEMATICAL SANS-SERIF ITALIC SMALL J
+  "\uD835\uDE5E", "\uD835\uDE5F",   // MATHEMATICAL SANS-SERIF BOLD ITALIC SMALL I..MATHEMATICAL SANS-SERIF BOLD ITALIC SMALL J
+  "\uD835\uDE92", "\uD835\uDE93",   // MATHEMATICAL MONOSPACE SMALL I..MATHEMATICAL MONOSPACE SMALL J
+];
+assert.sameValue(Soft_Dotted.length, 46, "Total code points with Soft_Dotted property");
+
+function charInfo(ch) {
+  function hexString(n) {
+    var s = n.toString(16).toUpperCase();
+    return "0000".slice(s.length) + s;
+  }
+
+  if (ch.length === 1) {
+    return "U+" + hexString(ch.charCodeAt(0));
+  }
+  var high = ch.charCodeAt(0);
+  var low = ch.charCodeAt(1);
+  var codePoint = ((high << 10) + low) + (0x10000 - (0xD800 << 10) - 0xDC00);
+  return "U+" + hexString(codePoint) + " = " + hexString(high) + " " + hexString(low);
+}
+
+
+// COMBINING DOT ABOVE (U+0307) removed when preceded by Soft_Dotted.
+// Character directly preceded by Soft_Dotted.
+assert.sameValue(
+  "i\u0307".toLocaleUpperCase("lt"),
+  "I",
+  "COMBINING DOT ABOVE preceded by LATIN SMALL LETTER I"
+);
+assert.sameValue(
+  "j\u0307".toLocaleUpperCase("lt"),
+  "J",
+  "COMBINING DOT ABOVE preceded by LATIN SMALL LETTER J"
+);
+assert.sameValue(
+  "\u012F\u0307".toLocaleUpperCase("lt"),
+  "\u012E",
+  "COMBINING DOT ABOVE preceded by LATIN SMALL LETTER I WITH OGONEK (U+012F)"
+);
+assert.sameValue(
+  "\u1ECB\u0307".toLocaleUpperCase("lt"),
+  "\u1ECA",
+  "COMBINING DOT ABOVE preceded by LATIN SMALL LETTER I WITH DOT BELOW (U+1ECB)"
+);
+assert.sameValue(
+  "\uD835\uDC22\u0307".toLocaleUpperCase("lt"),
+  "\uD835\uDC22",
+  "COMBINING DOT ABOVE preceded by MATHEMATICAL BOLD SMALL I (U+1D422 = D835 DC22)"
+);
+for (var i = 0; i < Soft_Dotted.length; ++i) {
```

Comment:
Yes, they are. I think I started with the explicit tests and only later added the `for` loop. I've removed them in the fixup.